avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerializer.java

```diff
@@ -112,7 +114,16 @@ protected byte[] serializeImpl(
       int id;
       if (autoRegisterSchema) {
         restClientErrorMsg = "Error registering Avro schema";
-        id = schemaRegistry.register(subject, schema, normalizeSchema);
+        io.confluent.kafka.schemaregistry.client.rest.entities.Schema s =
+            registerWithResponse(subject, schema, normalizeSchema);
+        if (s.getSchema() != null) {
+          Optional<ParsedSchema> optSchema = schemaRegistry.parseSchema(s);
+          if (optSchema.isPresent()) {
+            schema = (AvroSchema) optSchema.get();
+            schema = schema.copy(s.getVersion());
```

Comment:
why using 2 lines with `schema = ` here? will the latter one overwrite the former one?

Reply:
Yes, the second line just sets the version from `s`